specification/containerservice/resource-manager/Microsoft.ContainerService/preview/2022-05-02-preview/managedClusters.json

```diff
@@ -6059,6 +6059,24 @@
         "keyId": {
           "type": "string",
           "description": "Identifier of Azure Key Vault key. See [key identifier format](https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#vault-name-and-object-name) for more details. When Azure Key Vault key management service is enabled, this field is required and must be a valid key identifier. When Azure Key Vault key management service is disabled, leave the field empty."
+        },
+        "keyVaultNetworkAccess": {
+          "type": "string",
+          "enum": [
+            "Public",
+            "Private"
+          ],
+          "default": "Public",
+          "x-ms-enum": {
+            "name": "KeyVaultNetworkAccess",
```

Comment:
Better to distinguish it from the property name. Maybe something like `KeyVaultNetworkAccessTypes`.

Reply:
Sure. What's the purpose of "x-ms-enum"?

Reply:
See [details](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/creating-swagger.md#enums-with-x-ms-enum) of "x-ms-enum".

One example would be the following snippet from Python SDK, the name in x-ms-enum would be the class name.
```
class TrustedAccessRoleBindingProvisioningState(with_metaclass(CaseInsensitiveEnumMeta, str, Enum)):
    """The current provisioning state of trusted access role binding.
    """

    SUCCEEDED = "Succeeded"
    FAILED = "Failed"
    UPDATING = "Updating"
    DELETING = "Deleting"
```